### ISSUE TITLE
Type error is not detected when in while loop

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#2574]
The code below generates a runtime error because `non_property` was not decorated with `@property`. The expression `a - 0` should result in a type error, but it doesn't.

```python
class Foo:
    def non_property(self) -> int:
        ...

    def do_stuff(self, x: int | None):
        while True:
            if x is not None:
                a = x
            else:
                a = self.non_property

            _ = a - 0
```

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.188, which I just published. It will also be included in the next release of pylance.
